include/boost/geometry/algorithms/detail/overlay/get_turn_info_for_endpoint.hpp

```diff
@@ -497,13 +467,12 @@ struct get_turn_info_for_endpoint
                 }
                 else if ( ip_j2 )
                 {
-                    side_calculator_for_endpoint<cs_tag, RobustPoint1, RobustPoint2,
-                                    typename IntersectionInfo::side_strategy_type,
-                                    RobustPoint2>
-                        side_calc(ri2, rj1, ri1, ri2, rj2, rk2, inters.get_side_strategy()); // TODO: wrong, ri2/ri1/rj2
-                    
+                    int const side_pi_q2 = sides.apply(range2.at(1), range2.at(2), range1.at(0));
+                    int const side_pi_x = sides.apply(range2.at(0), range1.at(1), range1.at(0));
+                    int const side_qk_x = sides.apply(range2.at(0), range1.at(1), range2.at(2));
+
                     std::pair<operation_type, operation_type>
-                        operations = operations_of_equal(side_calc);
+                        operations = operations_of_equal(side_pi_q2, side_pi_x, side_qk_x);
```

Comment:
So unless we know what's happening exactly we should probably be careful about changing this code. But if all of the tests pass with this change it should be ok I think.